django/urls/resolvers.py

```diff
@@ -737,8 +759,16 @@ def _reverse_with_prefix(self, lookup_view, _prefix, *args, **kwargs):
                 else:
                     if set(kwargs).symmetric_difference(params).difference(defaults):
                         continue
-                    if any(kwargs.get(k, v) != v for k, v in defaults.items()):
+                    matches = True
```

Comment:
I think this can be rewritten ala:
```
for k,v in defaults.items():
   …
   if kwargs..get(k,v) != v:
      break
else:
  continue
```

Reply:
actually, we can't, the thing is that
```
for k,v in defaults.items():
   …
   if kwargs..get(k,v) != v:  # if this statment is True -- Statement 1
      matches = False
      break
if not matches:  # then this statement is also True -- Statement 2
  continue

```

both are not working opposite to each other, rather one is supporting the functioning of the other.

Reply:
Ah yeah, I think I missed the `not` in the if.